src/main/java/net/minestom/server/network/packet/PacketVanilla.java

```diff
@@ -0,0 +1,39 @@
+package net.minestom.server.network.packet;
+
+import net.minestom.server.MinecraftServer;
+import net.minestom.server.ServerFlag;
+import net.minestom.server.network.ConnectionState;
+import net.minestom.server.network.NetworkBuffer;
+import net.minestom.server.network.packet.client.ClientPacket;
+import net.minestom.server.network.packet.client.configuration.ClientFinishConfigurationPacket;
+import net.minestom.server.network.packet.client.handshake.ClientHandshakePacket;
+import net.minestom.server.network.packet.client.login.ClientLoginAcknowledgedPacket;
+import net.minestom.server.network.packet.server.ServerPacket;
+import net.minestom.server.utils.ObjectPool;
+
+/**
+ * Constants and utilities for vanilla packets.
+ */
+public final class PacketVanilla {
+    public static final PacketParser<ClientPacket> CLIENT_PACKET_PARSER = new PacketParser.Client();
+    public static final PacketParser<ServerPacket> SERVER_PACKET_PARSER = new PacketParser.Server();
+
+    /**
+     * Pool containing a buffer able to hold the largest packet.
+     */
+    public static final ObjectPool<NetworkBuffer> PACKET_POOL = ObjectPool.pool(
```

Comment:
Why make these always max size? You could just let them resize up from a much smaller size, and then assuming you never send a max size packet they would generally stay smaller no? I think most people never send particularly huge packets.